src/pages/UserProfile/UserProfile.tsx

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { Layout } from '../../components/Layout/Layout';
+import placeholder_club from '../../assets/images/placeholder_club.png';
+import { MemberType } from '../../model/Member';
+import { getMemberDetail, updateMember } from '../../api/membersApi';
+import { MyClubsCard } from '../../components/userprofile/MyClubsCard/MyClubsCard';
+import { UserProfileInfo } from '../../components/userprofile/UserProfileInfo/UserProfileInfo';
+import { useAuth } from '../../context/AuthContext';
+import { MemberCurrentlyReadingCard } from '../../components/userprofile/MembersCurrentlyReadingCard/MemberCurrentlyReadingCard';
+import { NotFound } from '../../components/NotFound/NotFound';
+import { Loading } from '../../components/Loading/Loading';
+import style from './UserProfile.module.scss';
+
+export const UserProfile = () => {
+  const { id } = useParams<{ id: string }>();
+  const user = useAuth();
+
+  const [memberDetail, setMemberDetail] = useState<MemberType | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const isOwner = user?.id === memberDetail?.id;
+
+  useEffect(() => {
+    const getMemberData = async () => {
+      setIsLoading(true);
+      if (id) {
+        const membersData = await getMemberDetail(id);
+        setMemberDetail(membersData);
+      }
+      setIsLoading(false);
```

Comment:
Dont be afraid of the new lines. Separate stuff. We dont need a blob of code
```suggestion
      setIsLoading(true);
      
      if (id) {
        const membersData = await getMemberDetail(id);
        setMemberDetail(membersData);
      }
      
      setIsLoading(false);
```